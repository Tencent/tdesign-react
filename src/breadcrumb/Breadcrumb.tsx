import React from 'react';
import classNames from 'classnames';
import useConfig from '../hooks/useConfig';
import forwardRefWithStatics from '../_util/forwardRefWithStatics';
import BreadcrumbItem from './BreadcrumbItem';
import { BreadcrumbProps } from './BreadcrumbProps';
import { BreadcrumbContext } from './BreadcrumbContext';
import { breadcrumbDefaultProps } from './defaultProps';

const Breadcrumb = forwardRefWithStatics(
  (props: BreadcrumbProps, ref) => {
    const { children, options, separator, maxItemWidth, className, ...restProps } = props;
    const { classPrefix } = useConfig();

    let content = children;
    if (options && options.length) {
      content = options.map((option, index) => (
        <BreadcrumbItem
          key={index}
          maxWidth={option.maxWidth}
          disabled={option.disabled}
          href={option.href}
          target={option.target}
          to={option.to}
          router={option.router}
          replace={option.replace}
          separator={separator}
          maxItemWidth={maxItemWidth}
        >
          {option.content || option.children}
        </BreadcrumbItem>
      ));
    }

    return (
      <BreadcrumbContext.Provider
        value={{
          maxItemWidthInContext: maxItemWidth,
          separator,
        }}
      >
        <div ref={ref} className={classNames(`${classPrefix}-breadcrumb`, className)} {...restProps}>
          {content}
        </div>
      </BreadcrumbContext.Provider>
    );
  },
  { BreadcrumbItem },
);

Breadcrumb.displayName = 'Breadcrumb';
Breadcrumb.defaultProps = breadcrumbDefaultProps;

export default Breadcrumb;
