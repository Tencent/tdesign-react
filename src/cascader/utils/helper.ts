import { TreeNode, CascaderContextType, TreeNodeValue, TreeOptionData } from '../interface';

/**
 * 获取node的全部路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}

/**
 * treeValue计算方法
 * @param value
 * @returns
 */
export const getTreeValue = (value: CascaderContextType['value']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && typeof value[0] === 'object') {
      treeValue = (value as TreeOptionData[]).map((val) => val.value);
    } else if (value.length) {
      treeValue = value as TreeNodeValue[];
    }
  } else if (value) {
    if (typeof value === 'object') {
      treeValue = [(value as TreeOptionData).value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};

/**
 * 判断node是否在页面上
 * @param node Node
 */
export function isInPage(node: Node) {
  return node === document.body ? false : document.body.contains(node);
}

export default {
  getFullPathLabel,
};
