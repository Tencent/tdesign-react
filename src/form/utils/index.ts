import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
