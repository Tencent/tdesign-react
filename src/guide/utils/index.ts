import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
