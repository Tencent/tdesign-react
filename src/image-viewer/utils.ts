export const downloadFile = function (imgSrc) {
  const image = new Image();
  const name = imgSrc?.split?.('/').pop() || Math.random().toString(32).slice(2);

  image.setAttribute('crossOrigin', 'anonymous');

  image.onload = () => {
    const canvas = document.createElement('canvas');
    canvas.width = image.width;
    canvas.height = image.height;

    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0, image.width, image.height);
    canvas.toBlob((blob) => {
      const url = URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.download = name;
      a.href = url;
      a.click();
      a.remove();
      URL.revokeObjectURL(url);
    });
  };
  image.src = imgSrc;
};
