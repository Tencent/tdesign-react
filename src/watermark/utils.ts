const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
