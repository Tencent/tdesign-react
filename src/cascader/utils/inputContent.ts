import isFunction from 'lodash/isFunction';
import isEmpty from 'lodash/isEmpty';
import { TreeNode, CascaderContextType, TreeNodeValue, CascaderProps } from '../interface';
import { isInPage } from './helper';

/**
 * icon Class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getCloseIconClass(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { visible, disabled } = cascaderContext;
  return [
    `${prefix}-cascader__icon`,
    {
      [CLASSNAMES.STATUS.visible]: visible,
      [CLASSNAMES.STATUS.disabled]: disabled,
    },
  ];
}

/**
 * icon Class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getFakeArrowIconClass(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { disabled } = cascaderContext;
  return [
    `${prefix}-cascader__icon`,
    {
      [CLASSNAMES.STATUS.disabled]: disabled,
    },
  ];
}

/**
 * Cascader外显div class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getCascaderInnerClasses(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { disabled, visible, size, multiple } = cascaderContext;
  return [
    `${prefix}-cascader`,
    {
      [CLASSNAMES.STATUS.disabled]: disabled,
      [CLASSNAMES.STATUS.active]: visible,
      [CLASSNAMES.SIZE[size]]: size,
      [`${prefix}-cascader--multiple`]: multiple,
    },
  ];
}

/**
 * closeIcon 是否显示
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getCloseShow(isHover: boolean, cascaderContext: CascaderContextType) {
  const { value, disabled, clearable, visible } = cascaderContext;
  return !!(!visible && clearable && isHover && !disabled && !isEmpty(value));
}

/**
 * Placeholder 是否显示
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getPlaceholderShow(
  cascaderContext: CascaderContextType,
  singleContent: string,
  multipleContent: TreeNode[],
) {
  const { filterable, visible, multiple } = cascaderContext;
  const valEmpty = !(multiple ? multipleContent.length : singleContent);
  if (filterable) {
    return valEmpty && !visible;
  }
  return valEmpty;
}

/**
 * 单选状态下内容
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getSingleContent(cascaderContext: CascaderContextType) {
  const { value, multiple, treeStore, showAllLevels } = cascaderContext;
  if (multiple || !value) return '';
  if (Array.isArray(value)) return '';
  const node = treeStore && treeStore.getNodes(value as TreeNodeValue | TreeNode);
  if (!(node && node.length)) {
    // 无匹配数据直接输出
    return value as string;
  }
  const path = node && node[0].getPath();
  if (path && path.length) {
    return showAllLevels ? path.map((node: TreeNode) => node.label).join(' / ') : path[path.length - 1].label;
  }
  return value as string;
}

/**
 * 多选状态下选中内容
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getMultipleContent(cascaderContext: CascaderContextType) {
  const { value, multiple, treeStore } = cascaderContext;

  if (!multiple) return [];
  if (multiple && !Array.isArray(value)) return [];

  const node = treeStore && treeStore.getNodes(value as TreeNodeValue | TreeNode);
  if (!node) return [];

  const path = (value as TreeNodeValue[]).map((item: TreeNodeValue) => {
    const node = treeStore.getNodes(item);
    return node[0];
  });
  if (value && (value as TreeNodeValue[]).length) {
    return path && path.length ? path : (value as TreeNode[]);
  }
  return [];
}

/**
 * 点击popup outside副作用
 * @param ref
 * @param cascaderContext
 * @param event
 * @returns
 */
export function outerClickListenerEffect(
  ref: HTMLElement,
  cascaderContext: CascaderContextType,
  event: MouseEvent | TouchEvent,
) {
  const { visible, setVisible } = cascaderContext;
  const target = event.target as Node;
  // 可能点击的dom在判断的时候已经被移除了
  if (!visible || !ref || ref.contains(target) || !isInPage(target)) return;
  setVisible(false);
}

/**
 * closeIcon点击副作用
 * @param cascaderContext
 */
export function closeIconClickEffect(cascaderContext: CascaderContextType) {
  const { setVisible, multiple, setExpend, setValue } = cascaderContext;

  setVisible(false);

  // 手动设置的展开需要去除
  if (multiple) {
    setExpend([]);
  }

  setValue(multiple ? [] : '', 'clear');
}

/**
 * tag 关闭按钮点击副作用
 * @param cascaderContext
 */
export function handleRemoveTagEffect(
  cascaderContext: CascaderContextType,
  node: TreeNode,
  onRemove: CascaderProps['onRemove'],
) {
  const { disabled, setValue } = cascaderContext;

  if (disabled) return;
  const checked = node.setChecked(!node.isChecked());
  setValue(checked, 'uncheck', node.getModel());
  if (isFunction(onRemove)) {
    onRemove({ value: checked, node: node as any });
  }
}

/**
 * 外显content点击副作用
 * @param cascaderContext
 */
export function innerContentClickEffect(cascaderContext: CascaderContextType) {
  const { setVisible, visible, filterActive, disabled } = cascaderContext;

  if (!disabled) {
    if (visible && filterActive) return; // input filter is active
    setVisible(!visible);
  }
}
