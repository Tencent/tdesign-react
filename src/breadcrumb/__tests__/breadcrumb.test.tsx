import React from 'react';
import { vi, render } from '@test/utils';
import Breadcrumb from '../Breadcrumb';

const { BreadcrumbItem } = Breadcrumb;

describe('Breadcrumb', () => {
  const rootTestID = 'breadcrumbRoot';
  const childTestID = 'breadcrumbChild';

  test('trigger events on breadcrumbItem', () => {
    const mockFn = vi.fn();
    const el = (
      <Breadcrumb>
        <BreadcrumbItem onClick={mockFn} data-testid={childTestID}>
          hello
        </BreadcrumbItem>
      </Breadcrumb>
    );
    const wrapper = render(el);

    const child = wrapper.getByTestId(childTestID);
    child.click();

    expect(mockFn).toHaveBeenCalled();
  });

  test('use custom separator', () => {
    const el = (
      <Breadcrumb data-testid={rootTestID}>
        <BreadcrumbItem separator={<span role="separator"> xxx </span>} data-testid={childTestID} />
      </Breadcrumb>
    );
    const wrapper = render(el);

    const root = wrapper.getByTestId(rootTestID);
    const separator = wrapper.getByRole('separator');

    expect(root).toContainElement(separator);
  });

  test('render options breadCrumbItem correctly', () => {
    const contentShouldNotBeExist = 'hello';
    const options = [{ content: '页面1' }, { content: '页面2' }, { content: '页面3' }];
    const el = (
      <Breadcrumb options={options} data-testid={rootTestID}>
        <BreadcrumbItem>{contentShouldNotBeExist}</BreadcrumbItem>
      </Breadcrumb>
    );
    const wrapper = render(el);

    const root = wrapper.getByTestId(rootTestID);

    expect(root.childNodes).toHaveLength(options.length);
    expect(root).not.toHaveTextContent(contentShouldNotBeExist);
    root.childNodes.forEach((node, index) => {
      expect(node).toHaveTextContent(options[index]?.content);
    });
  });
});
