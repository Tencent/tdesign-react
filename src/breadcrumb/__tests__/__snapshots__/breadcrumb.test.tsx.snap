// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-breadcrumb"
    theme="light"
  >
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 200px;"
        >
          页面1
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 200px;"
        >
          页面2页面2页面2页面2页面2页面2页面2页面2
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          页面3页面3页面3页面3页面3页面3页面3页面3页面3页面3页面3页面3页面3页面3页面3
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`custom.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-breadcrumb"
  >
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 150;"
        >
          页面1
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        &gt;&gt;
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 150;"
        >
          页面2
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        &gt;&gt;
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 160;"
        >
          页面3
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        &gt;&gt;
      </span>
    </div>
  </div>
  <div
    class="t-breadcrumb"
  >
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 150;"
        >
          页面1
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        /////
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 150;"
        >
          页面2
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        /////
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 160;"
        >
          页面3
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        /////
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`href.jsx 1`] = `
<DocumentFragment>
  <div>
    <div
      class="t-breadcrumb"
    >
      <div
        class="t-breadcrumb__item"
      >
        <a
          class="t-breadcrumb--text-overflow t-link"
          href="/"
          target="_blank"
        >
          <span
            class="t-breadcrumb__inner"
            style="max-width: 150;"
          >
            页面1
          </span>
        </a>
        <span
          class="t-breadcrumb__separator"
        >
          <svg
            class="t-icon t-icon-chevron-right"
            fill="none"
            height="1em"
            style="color: rgba(0, 0, 0, 0.3);"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
      </div>
      <div
        class="t-breadcrumb__item"
      >
        <span
          class="t-breadcrumb--text-overflow t-is-disabled"
        >
          <span
            class="t-breadcrumb__inner"
            style="max-width: 150;"
          >
            页面2
          </span>
        </span>
        <span
          class="t-breadcrumb__separator"
        >
          <svg
            class="t-icon t-icon-chevron-right"
            fill="none"
            height="1em"
            style="color: rgba(0, 0, 0, 0.3);"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
      </div>
      <div
        class="t-breadcrumb__item"
      >
        <span
          class="t-breadcrumb--text-overflow"
        >
          <span
            class="t-breadcrumb__inner"
            style="max-width: 160;"
          >
            页面3
          </span>
        </span>
        <span
          class="t-breadcrumb__separator"
        >
          <svg
            class="t-icon t-icon-chevron-right"
            fill="none"
            height="1em"
            style="color: rgba(0, 0, 0, 0.3);"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      style="margin-top: 20px;"
    >
      点击计数器：0
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-breadcrumb"
  >
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          <svg
            class="t-icon t-icon-books"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M8 9.28l-4 2.8V3h8v9.08l-4-2.8zm0 1.22l4.21 2.95a.5.5 0 00.79-.41V3a1 1 0 00-1-1H4a1 1 0 00-1 1v10.04c0 .4.45.64.79.4l4.2-2.94z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
          页面1
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          <svg
            class="t-icon t-icon-books"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M8 9.28l-4 2.8V3h8v9.08l-4-2.8zm0 1.22l4.21 2.95a.5.5 0 00.79-.41V3a1 1 0 00-1-1H4a1 1 0 00-1 1v10.04c0 .4.45.64.79.4l4.2-2.94z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
          页面2
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          <svg
            class="t-icon t-icon-books"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M8 9.28l-4 2.8V3h8v9.08l-4-2.8zm0 1.22l4.21 2.95a.5.5 0 00.79-.41V3a1 1 0 00-1-1H4a1 1 0 00-1 1v10.04c0 .4.45.64.79.4l4.2-2.94z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
          页面3
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`options.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-breadcrumb"
  >
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          页面1
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          页面2
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <a
        class="t-breadcrumb--text-overflow t-link"
        href="https://github.com/"
        target="_self"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          页面3
        </span>
      </a>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`to.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-breadcrumb"
  >
    <div
      class="t-breadcrumb__item"
    >
      <a
        class="t-breadcrumb--text-overflow t-link"
        target="_self"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          首页
        </span>
      </a>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <a
        class="t-breadcrumb--text-overflow t-link"
        target="_self"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 120px;"
        >
          页面1
        </span>
      </a>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`width.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-breadcrumb"
  >
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 100px;"
        >
          父级设置100px父级设置100px
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 160px;"
        >
          设置最大宽度160px设置最大宽度160px设置最大宽度160px
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 240px;"
        >
          设置最大宽度240px设置最大宽度240px设置最大宽度240px设置最大宽度240px设置最大宽度240px
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
    <div
      class="t-breadcrumb__item"
    >
      <span
        class="t-breadcrumb--text-overflow"
      >
        <span
          class="t-breadcrumb__inner"
          style="max-width: 100px;"
        >
          父级设置100px父级设置100px
        </span>
      </span>
      <span
        class="t-breadcrumb__separator"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          style="color: rgba(0, 0, 0, 0.3);"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
